be/src/vec/data_types/data_type_factory.cpp

```diff
@@ -260,4 +260,67 @@ DataTypePtr DataTypeFactory::create_data_type(const PColumnMeta& pcolumn) {
     return nested;
 }
 
+DataTypePtr DataTypeFactory::create_data_type(const arrow::Type::type& type, bool is_nullable) {
```

Comment:
Please pass arrow::DataType as input args to avoid refractor later, because complex types create need children info.